cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -490,12 +490,117 @@ def referenced_views_and_containers_are_existing(self) -> "DMSRules":
             raise issues.MultiValueError(errors)
         return self
 
+    @model_validator(mode="after")
+    def validate_extension(self) -> "DMSRules":
+        if self.metadata.schema_ is not SchemaCompleteness.extended:
+            return self
+        if not self.reference:
+            raise ValueError("The schema is set to 'extended', but no reference rules are provided to validate against")
+        is_solution = self.metadata.space != self.reference.metadata.space
+        if is_solution:
+            return self
+        if self.metadata.extension is ExtensionCategory.rebuild:
+            # Everything is allowed
+            return self
+        # Is an extension of an existing model.
+        user_schema = self.as_schema()
+        ref_schema = self.reference.as_schema()
+        new_containers = {container.as_id(): container for container in user_schema.containers}
+        existing_containers = {container.as_id(): container for container in ref_schema.containers}
```

Comment:
we should consider updating base structure for all our sheets (except metdata), as currently they do not provide mechanism for fast accesses via keys to specific "rows". Rules v1 , we have access through "key", which is a unique ID of a thing (e.g. property, or class). I often find myself doing the same as you do here (rebuild a data structure for easy access)

Reply:
Current structure allows for repetition and redefinition